utils/zerovec/src/varzerovec/components.rs

```diff
@@ -28,40 +27,60 @@ pub(super) const MAX_INDEX: usize = u32::MAX as usize;
 ///
 /// Do not implement this trait, its internals may be changed in the future,
 /// and all of its associated items are hidden from the docs.
+pub trait VarZeroVecFormat: 'static + Sized {
+    /// The type to use for the indexing array
+    ///
+    /// Safety: must be a ULE for which all byte sequences are allowed
+    #[doc(hidden)]
+    type Index: IntegerULE;
+    /// The type to use for the length segment
+    ///
+    /// Safety: must be a ULE for which all byte sequences are allowed
+    #[doc(hidden)]
+    type Len: IntegerULE;
+}
+
+/// This trait represents various ULE types that can be used to represent an integer
+///
+/// Do not implement this trait, its internals may be changed in the future,
+/// and all of its associated items are hidden from the docs.
 #[allow(clippy::missing_safety_doc)] // no safety section for you, don't implement this trait period
-pub unsafe trait VarZeroVecFormat: 'static + Sized {
+#[doc(hidden)]
+pub unsafe trait IntegerULE: ULE {
     /// The error to show when unable to construct a vec
     #[doc(hidden)]
     const TOO_LARGE_ERROR: &'static str;
+
+    /// Safety: must be sizeof(self)
```

Comment:
Suggestion: Why not just use size_of::<Self>() which is const?

Reply:
I found it makes the code easier to read but I'm open to that being a followup

Reply:
basically `mem::size_of::<F::Index>` feels clunkier than `F::Index::SIZE` in the situations it gets used in.

_however_ in my next PR I'm removing METADATA_WIDTH and that may make the expressions simple enough that a sizeof won't be a big deal